internal/daemon/vendor-specific-plugins/intel-netsec/main.go

```diff
@@ -0,0 +1,367 @@
+	isDPUMode      bool
+	dpuPcieAddress string
+}
+
+func SetSriovNumVfs(pciAddr string, numVfs int) error {
+	klog.Infof("SetSriovNumVfs(): set NumVfs device %s numvfs %d", pciAddr, numVfs)
+	numVfsFilePath := filepath.Join("/sys/bus/pci/devices", pciAddr, "sriov_numvfs")
+	bs := []byte(strconv.Itoa(numVfs))
+	err := os.WriteFile(numVfsFilePath, []byte("0"), os.ModeAppend)
+	if err != nil {
+		klog.Errorf("SetSriovNumVfs(): fail to reset NumVfs file path %s, err %v", numVfsFilePath, err)
+		return err
+	}
+	if numVfs == 0 {
+		return nil
+	}
+	err = os.WriteFile(numVfsFilePath, bs, os.ModeAppend)
+	if err != nil {
+		klog.Errorf("SetSriovNumVfs(): fail to set NumVfs file path %s, err %v", numVfsFilePath, err)
+		return err
+	}
+	return nil
+}
+
+func (vsp *intelNetSecVspServer) GetVFs(pfPCIeAddress string) ([]string, error) {
+	var pciVFAddresses []string
+
+	pciInfo, err := ghw.PCI()
+	if err != nil {
+		return nil, err
+	}
+
+	bus := ghw.PCIAddressFromString(pfPCIeAddress).Bus
+	for _, pci := range pciInfo.Devices {
+		if ghw.PCIAddressFromString(pci.Address).Bus == bus {
+			if pci.Vendor.ID == IntelVendorID &&
+				pci.Product.ID == IntelNetSecHostVfDeviceID {
+				pciVFAddresses = append(pciVFAddresses, pci.Address)
+			}
+		}
+	}
+
+	numVfs := len(pciVFAddresses)
+	vsp.log.Info("GetVFs(): found VFs", "NumVFs", numVfs, "DpuPcieAddress", vsp.dpuPcieAddress)
+	return pciVFAddresses, nil
+}
+
+func (vsp *intelNetSecVspServer) GetNetDevNameFromPCIeAddr(pcieAddress string) string {
+	netInfo, err := ghw.Network()
+	if err != nil {
+		vsp.log.Error(err, "GetNetDevNameFromPCIeAddr(): failed to get network info")
+		return ""
+	}
+
+	for _, nic := range netInfo.NICs {
+		if nic.PCIAddress != nil && *nic.PCIAddress == pcieAddress {
+			vsp.log.Info("GetNetDevNameFromPCIeAddr(): found DPU network device", "Name", nic.Name, "PCIAddress", *nic.PCIAddress)
+			return nic.Name
+		}
+	}
+
+	vsp.log.Error(nil, "GetNetDevNameFromPCIeAddr(): Network device not found", "PCIAddress", pcieAddress)
+	return ""
+}
+
+func linkHasAddrgenmodeEui64(interfaceName string) bool {
+	out, err := exec.Command("ip", "-d", "link", "show", "dev", interfaceName).Output()
+	return err == nil && strings.Contains(string(out), "addrgenmode eui64")
+}
+
+func enableIPV6LinkLocal(interfaceName string, ipv6Addr string) error {
+	// Tell NetworkManager to not manage our interface.
+	err1 := exec.Command("nsenter", "-t", "1", "-m", "-u", "-n", "-i", "--", "nmcli", "device", "set", interfaceName, "managed", "no").Run()
+	if err1 != nil {
+		// This error may be fine. Maybe our host doesn't even run
+		// NetworkManager. Ignore.
+		klog.Infof("nmcli device set %s managed no failed with error %v", interfaceName, err1)
+	}
+
+	optimistic_dad_file := "/proc/sys/net/ipv6/conf/" + interfaceName + "/optimistic_dad"
+	err1 = os.WriteFile(optimistic_dad_file, []byte("1"), os.ModeAppend)
+	if err1 != nil {
+		klog.Errorf("Error setting %s: %v", optimistic_dad_file, err1)
+	}
```

Comment:
This could be a function with a name like disableDuplicateAddressDetection

Reply:
How was this done without the DPU operator? Do we ened this only for the vsp link?

Reply:
I am reusing logic from Marvell which I will clean up and make it a library. It is used for the VSP link.